suite-native/module-connect-device/src/components/PinFormControlButtons.tsx

```diff
@@ -27,23 +32,47 @@ type NavigationProp = StackToStackCompositeNavigationProps<
 >;
 
 export const PinFormControlButtons = () => {
+    const [isPinVerified, setIsPinVerified] = useState(false);
+    const dispatch = useDispatch();
+
     const openLink = useOpenLink();
     const { translate } = useTranslate();
     const navigation = useNavigation<NavigationProp>();
-    const isDeviceUnlocked = useSelector(selectIsDeviceUnlocked);
     const { showAlert } = useAlert();
     const { handleSubmit, getValues, watch, setValue, reset } = useFormContext();
+    const device = useSelector(selectDevice) ?? null;
+    const hasDeviceAuthFailed = useSelector(selectDeviceAuthFailed);
+    const isDeviceReadyToUseAndAuthorized = useSelector(selectIsDeviceReadyToUseAndAuthorized);
+
+    const handleDeviceChange = useCallback(() => {
+        if (hasDeviceAuthFailed) {
+            reset();
+        } else {
+            // Do not navigate out of the PIN screen right away, wait for isDeviceReadyToUseAndAuthorized
+            setIsPinVerified(true);
+        }
+    }, [hasDeviceAuthFailed, reset]);
 
     useEffect(() => {
-        // Connect doesn't have an event for correct pin entry,
-        // but we can subscribe to this property change instead
-        // which gets changed after succesfull auth
-        if (isDeviceUnlocked) {
-            navigation.navigate(RootStackRoutes.ConnectDevice, {
-                screen: ConnectDeviceStackRoutes.ConnectingDevice,
-            });
+        if (isPinVerified && isDeviceReadyToUseAndAuthorized) {
+            if (navigation.canGoBack()) {
+                navigation.goBack();
+            }
+            reset();
+            dispatch(
+                removeButtonRequests({
+                    device,
+                    buttonRequestCode: 'PinMatrixRequestType_Current',
```

Comment:
ButtonRequest_PinEntry does not have to be removed?

Reply:
Nope, `ButtonRequest_PinEntry` is relevant for T2 only, and PinFormControlButtons is used only for T1.